### PR TITLE
fix rhv-upload function to work with keycloak instead of basic auth only

### PR DESCRIPTION
**Pull request by Schamane187**
As discusses here are a working change to be able  to use keycloak URL's like this.

https://admin@ovirt@internalsso@hostname/ovirt-engine/api

Thx again for the help and the quick fix

### CODE REVIEW

**Comment by rwmjones** [comment]
@nirs

File: output/rhv-upload-plugin.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 from contextlib import contextmanager
 from http.client import HTTPSConnection, HTTPConnection
-from urllib.parse import urlparse
+from urllib.parse import urlparse, urlunparse
```
Comment (nirs) [inline output/rhv-upload-plugin.py:28]: We don't need urlunparse() in this module, please revert this change.

This module uses ovirt-imageio url which cannot have a username in the netloc.

File: output/rhv-upload-precheck.py [hunk 2]
```diff
@@ -58,7 +59,7 @@
 
 # Connect to the server.
 connection = sdk.Connection(
-    url=params['output_conn'],
+    url=urlunparse(parsed._replace(netloc=netloc)),
```
Comment (nirs) [inline output/rhv-upload-precheck.py:62]: Need to remove the trailing whitespace.

**Review by nirs — COMMENTED** [review]
Thanks @Schamane187, looks good but need minor changes.

**Comment by nirs** [comment]
Need to squash before merging.

**Comment by rwmjones** [comment]
As Nir says please squash this into a single commit.
